src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,534 @@
+        logger.info("Executing command {} on {}".format(command, self.ip_address))
+        time.sleep(1)
+        ssh_stdin, ssh_stdout, ssh_stderr = self.ssh_connection.cmd(command)
+        time.sleep(1)
+        return ssh_stdout.readlines(), ssh_stderr.readlines()
+
+    def update_filter(self):
+        try:
+            logger.info("Updating filter")
+            # Check if the filters are already set
+            filter_cmd = '/usr/bin/crudini --get /etc/nova/nova.conf ' \
+                         'DEFAULT scheduler_default_filters'
+            filters = "RetryFilter,AvailabilityZoneFilter,RamFilter," \
+                      "DiskFilter,ComputeFilter,ComputeCapabilitiesFilter," \
+                      "ImagePropertiesFilter,ServerGroupAntiAffinityFilter," \
+                      "ServerGroupAffinityFilter,CoreFilter," \
+                      "NUMATopologyFilter," \
+                      "AggregateInstanceExtraSpecsFilter"
+            # Execute the command to check if filters already set
+            ssh_stdout_list, ssh_stderr_list = self.execute_cmd(filter_cmd)
+            if len(ssh_stdout_list) > 0:
+                if filters in ssh_stdout_list[0]:
+                    logger.warning("Nova schedular filters already exists on {} "
+                                   "skipping filters update.".format(self.ip_address))
+                    return
+
+            # Execute the update filter command
+            update_filter = '/usr/bin/crudini --set /etc/nova/nova.conf ' \
+                            'DEFAULT scheduler_default_filters ' + filters
+            ssh_stdout_list, ssh_stderr_list = self.execute_cmd(update_filter)
+            if ssh_stderr_list:
+                raise Exception("Command execution failed "
+                                "with error {}".format(ssh_stderr_list))
+            else:
+                # Execute the restart scheduler service command
+                sched_cmd = "/usr/bin/systemctl " \
+                            "restart openstack-nova-scheduler.service"
+                ssh_stdout_list, ssh_stderr_list = self.execute_cmd(sched_cmd)
+                time.sleep(2)
+
+                if ssh_stderr_list:
+                    logger.error("Failed to restart the scheduler service \
+                    with error".format(ssh_stderr_list))
+        except Exception as error:
+            message = "Exception {}: {}". \
+                      format(type(error).__name__, str(error))
+            logger.error("Failed to update filter on remote machine {} "
+                         "with error {}".format(self.ip_address, message))
+        finally:
+            # Close the ssh connection
+            self.ssh_connection.close()
+
+
+def edit_dell_environment_file(enable_hugepage,
+                               enable_numa,
+                               hugepage_size,
+                               vcpu_pin_set,
+                               dell_compute_count=0):
```

Comment:
Can you change the above line to not default to anything so that dell_compute_count is required?  It looks like it is only called from 1 place, and it passes the compute count.  A value of 0 for the compute count really is invalid.

Reply:
Agree. Removed, compute count  is at least one. It was needed at a time where we supposed it would have been possible to mix different type of compute (eg: regular + NFV ), meaning that our custom node count can be null.